crates/server/src/store/metrics.rs

```diff
@@ -17,10 +17,16 @@ impl StoreMetrics {
 
 pub(super) struct PersistentStoreMetrics {
     pub(super) zone_records: Gauge,
+    #[cfg(feature = "__dnssec")]
```

Comment:
Nit: squash this into the previous commit.

Reply:
It was requested to keep the history clear here. So I've split the commits into two.
https://github.com/hickory-dns/hickory-dns/pull/2900#discussion_r2031461591

Now I'm not sure how to handle that request. Please let me know which version is preferred.

Reply:
Fair enough, sorry for the unclear feedback. Just going to merge this now, thanks for all the work!